explorer/ast/statement.cpp

```diff
@@ -15,111 +15,136 @@ using llvm::cast;
 
 Statement::~Statement() = default;
 
-void Statement::PrintDepth(int depth, llvm::raw_ostream& out) const {
+void Statement::PrintDepth(int depth, int indent_num_spaces,
+                           llvm::raw_ostream& out) const {
   if (depth == 0) {
-    out << " ... ";
+    out.indent(indent_num_spaces) << " ... ";
     return;
   }
   switch (kind()) {
     case StatementKind::Match: {
       const auto& match = cast<Match>(*this);
-      out << "match (" << match.expression() << ") {";
+      out.indent(indent_num_spaces) << "match (" << match.expression() << ") {";
       if (depth < 0 || depth > 1) {
         out << "\n";
         for (const auto& clause : match.clauses()) {
-          out << "case " << clause.pattern() << " =>\n";
-          clause.statement().PrintDepth(depth - 1, out);
+          out.indent(indent_num_spaces + 2)
+              << "case " << clause.pattern() << " => ";
+          clause.statement().PrintDepth(depth - 1, indent_num_spaces + 2, out);
           out << "\n";
         }
       } else {
-        out << "...";
+        out.indent(indent_num_spaces) << "...";
       }
-      out << "}";
+      out.indent(indent_num_spaces) << "}";
       break;
     }
     case StatementKind::While: {
       const auto& while_stmt = cast<While>(*this);
-      out << "while (" << while_stmt.condition() << ")\n";
-      while_stmt.body().PrintDepth(depth - 1, out);
+      out.indent(indent_num_spaces)
+          << "while (" << while_stmt.condition() << ")\n";
+      if (depth < 0 || depth > 1) {
+        while_stmt.body().PrintDepth(depth - 1, indent_num_spaces + 2, out);
+      } else {
+        out.indent(indent_num_spaces) << "...";
+      }
       break;
     }
     case StatementKind::For: {
       const auto& for_stmt = cast<For>(*this);
-      out << "for (" << for_stmt.variable_declaration() << " in "
+      out.indent(indent_num_spaces)
+          << "for (" << for_stmt.variable_declaration() << " in "
           << for_stmt.loop_target() << ")\n";
-      for_stmt.body().PrintDepth(depth - 1, out);
+      if (depth < 0 || depth > 1) {
+        for_stmt.body().PrintDepth(depth - 1, indent_num_spaces + 2, out);
+      } else {
+        out.indent(indent_num_spaces) << "...";
+      }
       break;
     }
     case StatementKind::Break:
-      out << "break;";
+      out.indent(indent_num_spaces) << "break;";
       break;
     case StatementKind::Continue:
-      out << "continue;";
+      out.indent(indent_num_spaces) << "continue;";
       break;
     case StatementKind::VariableDefinition: {
       const auto& var = cast<VariableDefinition>(*this);
       if (var.is_returned()) {
-        out << "returned ";
+        out.indent(indent_num_spaces) << "returned ";
       }
-      out << "var " << var.pattern();
+      out.indent(var.is_returned() ? 0 : indent_num_spaces)
+          << "var " << var.pattern();
```

Comment:
Can you avoid the multiple `out.indent` calls if you call it before the `if (var.is_returned())`?

Reply:
Sure, I'll do it in the next commit.